tor_core/helpers.py

```diff
@@ -179,7 +179,53 @@ def update_wiki_page(pagename, content, config):
         )
 
 
-def run_until_dead(func, bot_name=None, exceptions=default_exceptions):
+def deactivate_heartbeat_port(port):
+    """
+    This isn't used as part of the normal functions; when a port is created,
+    it gets used again and again. The point of this function is to deregister
+    the port that the status page checks, but would probably only be used by
+    the command line.
+
+    :param port: int, the port number
+    :return: None
+    """
+    config.redis.srem('active_heartbeat_ports', port)
+    logging.info('Removed port from set of heartbeats.')
+
+
+def stop_heartbeat(config):
+    """
+    Any logic that goes along with stopping the cherrypy heartbeat server goes
+    here. This is called on exit of `run_until_dead()`, either through keyboard
+    or crash. The heartbeat server will terminate if the process dies anyway,
+    but this allows for a clean shutdown.
+
+    :param config: the global config object
+    :return: None
+    """
+    stop_heartbeat_server()
+    logging.info('Stopped heartbeat!')
+
+
+def run_until_dead(func, exceptions=default_exceptions):
+    """
+    The official method that replaces all that ugly boilerplate required to
+    start up a bot under the TranscribersOfReddit umbrella. This method handles
+    communication issues with Reddit, timeouts, and handles CTRL+C and
+    unexpected crashes.
+
+    :param func: the function that you want to run; this will automatically be
+        passed the config object. Historically, this is the only thing needed
+        to start a bot.
+    :param exceptions: A tuple of exception classes to guard against. These are
+        the praw connection exceptions by default; things like timeouts and
+        general connection errors, but they can be overridden with your own
+        set of exceptions if you feel it necessary.
```

Comment:
Seems pretty wordy for a description of the parameter. What about something more concise, such as:

> A tuple of exceptions to guard against (default: certain PRAW connection exceptions)